version: '3.4'

services:
  chatapp:
    container_name: ChatApp
    image: ${DOCKER_REGISTRY-}chatapp
    build:
      context: .
      dockerfile: ChatApp/Dockerfile
    depends_on:
        - postgres
    ports:
        - "5000:80"
    networks:
        - "chatapp-network"

  postgres:
    container_name: chatapp_postgres
    image: postgres:latest
    restart: on-failure:3
    ports:
        - "5432:5432"
    networks:
        - "chatapp-network"
    volumes:
        - postgres_chatapp_data:/var/lib/postgresql/data/
    environment:
        POSTGRES_USER: ${POSTGRES_USER}
        POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
        POSTGRES_DB: chatapp
        PGDATA: /var/lib/postgresql/data/

  pgadmin:
    container_name: pgadmin4_container
    image: dpage/pgadmin4
    restart: on-failure:3
    depends_on:
        - postgres
    environment:
        PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL}
        PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD}
        PGADMIN_LISTEN_PORT: "5050"
    ports:
      - "5050:5050"
    networks:
        - "chatapp-network"

networks:
    chatapp-network:
        driver: bridge

volumes:
    postgres_chatapp_data: